packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -0,0 +1,524 @@
+        }
+        headerTitle.insertAdjacentText('beforeend', group.label);
+        headerTitle.classList.add('header-title');
+        header.appendChild(headerTitle);
+        header.tabIndex = 0;
+        if (group.submenu) {
+            const submenuIcon = group.position === SmartConnectorPosition.Top ? createIcon('chevron-up') : createIcon('chevron-down');
+            header.appendChild(submenuIcon);
+            groupItems.classList.add(COLLAPSABLE_CLASS);
+            header.onclick = _ev => {
+                this.toggleSubmenu(submenuIcon, group, groupItems, searchField);
+            };
+            header.onkeydown = ev => {
+                if (matchesKeystroke(ev, 'Enter')) {
+                    this.toggleSubmenu(submenuIcon, group, groupItems, searchField);
+                }
+                this.headerNavigation(ev, groupItems, header);
+            };
+            this.groupIsCollapsed[group.id] = true;
+        }
+        return header;
+    }
+
+    protected toggleSubmenu(icon: HTMLElement, group: SmartConnectorGroupItem, groupItems: HTMLElement, searchField: HTMLElement): void {
+        changeCodiconClass(icon, 'chevron-up');
+        changeCodiconClass(icon, 'chevron-down');
+        this.groupIsCollapsed[group.id] = !this.groupIsCollapsed[group.id];
+        if (groupItems.style.maxHeight) {
+            groupItems.style.maxHeight = '';
+            searchField.style.maxHeight = '';
+        } else {
+            groupItems.style.maxHeight = MAX_HEIGHT_GROUP;
+            searchField.style.maxHeight = '50px';
+        }
+    }
+
+    protected createToolButton(item: PaletteItem): HTMLElement {
+        const button = document.createElement('div');
+        button.tabIndex = 0;
+        button.classList.add(TOOL_BUTTON_CLASS);
+        if (item.icon) {
+            button.appendChild(createIcon(item.icon));
+            return button;
+        }
+        button.insertAdjacentText('beforeend', item.label);
+        button.onclick = this.onClickCreateToolButton(button, item);
+        button.onkeydown = ev => this.toolButtonKeyHandler(ev, item);
+        button.id = item.id;
+        return button;
+    }
+
+    protected createSearchField(itemGroup: SmartConnectorGroupItem): HTMLElement {
+        const searchField = document.createElement('input');
+        searchField.classList.add('smart-connector-search');
+        searchField.id = itemGroup.id + SEARCH_FIELD_SUFFIX;
+        searchField.type = 'text';
+        searchField.placeholder = ' Search...';
+        searchField.onkeyup = () => this.requestFilterUpdate(this.searchFields[itemGroup.id].value, itemGroup);
+        searchField.onkeydown = ev => this.searchFieldKeyHandler(ev, itemGroup);
+        this.searchFields[itemGroup.id] = searchField;
+        const searchContainer = document.createElement('div');
+        const containerClass = itemGroup.submenu ? 'smart-connector-submenu-search-container' : 'smart-connector-search-container';
```

Comment:
Please extract those strings to the `consts` defined before as you already did in other places here.